webrtc/udp_connection.nim

```diff
@@ -0,0 +1,89 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import errors
+import chronos, chronicles
+
+logScope:
+  topics = "webrtc udp"
+
+# UdpConn is a small wrapper of the chronos DatagramTransport.
+# It's the simplest solution we found to store the message and
+# the remote address used by the underlying protocols (dtls/sctp etc...)
+
+type
+  WebRtcUdpError = object of WebRtcError
+
+  UdpPacketInfo* = tuple
+    message: seq[byte]
+    raddr: TransportAddress
+
+  UdpConn* = ref object
+    laddr*: TransportAddress
+    udp: DatagramTransport
+    dataRecv: AsyncQueue[UdpPacketInfo]
+    closed: bool
+
+proc init*(self: UdpConn, laddr: TransportAddress) =
```

Comment:
I think this is a constructor, isn't it? If so, it seems awkward to pass an instance to it. See more on https://status-im.github.io/nim-style-guide/language.objconstr.html